sotodlib/preprocess/preprocess_plot.py

```diff
@@ -0,0 +1,239 @@
+
+    for i, det_idx in enumerate(plot_dets_idx):
+        ax[0].errorbar(hwpss_stats.binned_angle, hwpss_stats.binned_signal[det_idx], yerr=hwpss_stats.sigma_bin[det_idx],
+                       alpha=0.5, color='tab:blue', fmt='.-', capsize=2, zorder=2, label='binned signal' if i ==0 else None)
+        ax[0].plot(aman.hwp_angle[:2000], aman.signal[det_idx][:2000]-np.median(aman.signal[det_idx][:2000]),
+                   alpha=0.5, color='tab:red', marker='o', markersize=0.5, linestyle='None', zorder=1,
+                   label='unbinned signal (2000 samps)' if i ==0 else None)
+
+        modes = [int(mode_name[1:]) for mode_name in list(hwpss_stats.modes.vals[::2])]
+        ax[0].plot(hwpss_stats.binned_angle, hwpss_stats.binned_model[det_idx], 
+                   alpha=0.9, color='tab:orange', zorder=3, label=f'binned model \n(modes = {modes})' if i ==0 else None)
+
+    ax[0].legend()
+    ax[0].set_xlabel('HWP angle [rad]')
+    ax[0].set_ylabel('Signal [Readout Radians]')
+    ax[0].set_title(f'random {plot_num_dets} detectors')
+
+    ax[1].hist(hwpss_stats.redchi2s, bins=np.logspace(start=-1, stop=2, num=50))
+    ax[1].axvline(x=np.nanmedian(hwpss_stats.redchi2s), linestyle='dashed', color='black',
+                  label=f'median: {np.nanmedian(hwpss_stats.redchi2s):.2f}')
+    ax[1].set_xscale('log')
+    ax[1].set_yscale('log')
+    ax[1].set_title(f'reduced chi2s distribution (Ndets={hwpss_stats.dets.count})')
+    ax[1].legend()
+
+    obs_ts = aman.timestamps[0]
+    det = aman.dets.vals[0]
+    plt.suptitle(f'HWPSS Stats for Obs_timestamp:{obs_ts:.0f}, dT = {np.ptp(aman.timestamps)/60:.1f} min\ndet:{det}\n')
+    plt.subplots_adjust(top=0.85, bottom=0.2)
+    head_tail = os.path.split(filename)
+    os.makedirs(head_tail[0], exist_ok=True)
+    plt.savefig(filename)
+
+def plot_sso_footprint(aman, planet_aman, sso, wafer_offsets=None, filename='./sso_footprint.png'):
+    """
+    Function for plotting SSO footprint.
+
+    Parameters
+    ----------
+    aman : AxisManager
+        Input axis manager.
+    planet_aman : AxisManager
+        Axis manager of results from obs_ops.sources.get_sso for a single planet.
+    sso : str
+        Name of planet.
+    wafer_offsets : dict
+        Dictionary of wafer offsets.
+    filename : str
+        Full filename with direct path to plot output directory.
+    """
+    xi_p = planet_aman['xi_p']
+    eta_p = planet_aman['eta_p']
+
+    if wafer_offsets is None:
+        # Default wafer offsets
+        wafer_offsets = {'ws0': (-2.5, -0.5),
+                         'ws1': (-2.5, -13),
+                         'ws2': (-13, -7),
+                         'ws3': (-13, 5),
+                         'ws4': (-2.5, 11.5),
+                         'ws5': (8.5, 5),
+                         'ws6': (8.5, -7)}
+
+    # Get default focal plane from sotodlib
+    hw = np.load('/so/home/msilvafe/shared_files/sat_hw_positions.npz')
```

Comment:
This file path needs to be an argument and if it's going to be used by the pipeline script needs to be stored somewhere more permanent than my user home directory.